src/message.rs

```diff
@@ -52,7 +54,7 @@ pub struct WebPushPayload {
     /// Encrypted content data.
     pub content: Vec<u8>,
     /// Headers depending on the authorization scheme and encryption standard.
-    pub crypto_headers: Vec<(&'static str, String)>,
+    pub crypto_headers: HashMap<String, String>,
```

Comment:
I guess the key allocations are necessary due to us using an external library. Typically header keys are known and we could save some allocations by using static slices. I guess it's not possible anymore? Also, hashing takes some CPU cycles. Are you sure you need it, or could we still just go with a vector of tuples?

Reply:
Yes, that's the "Breaking change" I was refering to. The `ece` crate exposes a HashMap and uses it internally, explaining the change. I could have made a conversion between the provided `HashMap<String,String>` but the result provided by the `ece` crate has a hashmap so it's already allocated and hashed there anyway.... The way I thought about it was : it's always going to be a very small hashmap, so the allocation and hashing should not be a major problem and the benefits of using the `ece` crate should outweight this problem.

Reply:
A compromise that could be achieve is: Getting back to a `Vec<&'a str, String>` somehow (that a static lifetime should satisfy), and converting the HashMap used for the `encrypt_aesgcm` function to such a Vector. Since the `encrypt` function for AES128GCM does not need it, we'd have allocation and hashing done only on a legacy protocol, and the up-to-date one (`aes128gcm`) which will most likely be preferred most of the time, would not be hindered by this.